src/datasets/arrow_writer.py

```diff
@@ -426,20 +426,15 @@ def write_examples_on_file(self):
         """Write stored examples from the write-pool of examples. It makes a table out of the examples and write it."""
         if not self.current_examples:
             return
-
-        # order the columns properly
+        # preserve the order the columns
         if self.schema:
             schema_cols = set(self.schema.names)
-            common_cols, extra_cols = [], []
-            for col in self.current_examples[0][0]:
-                if col in schema_cols:
-                    common_cols.append(col)
-                else:
-                    extra_cols.append(col)
+            examples_cols = self.current_examples[0][0].keys()  # .keys() preserves the order (unlike set)
+            common_cols = [col for col in self.schema.names if col in examples_cols]
+            extra_cols = [col for col in examples_cols if col not in schema_cols]
```

Comment:
What I was trying to say is the following, to have a speedup:

```suggestion
            examples_cols = self.current_examples[0][0].keys()  # .keys() preserves the order (unlike set)
            examples_cols_set = set(examples_cols)
            common_cols = [col for col in self.schema.names if col in examples_cols_set]
            extra_cols = [col for col in examples_cols if col not in schema_cols]
```

Reply:
There shouldn't be any difference in speed (when checking membership) between the built-in  `set` and the `keys` object.

Reply:
Oh, TIL!